### PR TITLE
Graph time spent doing environment check by app

### PR DESCRIPTION
**Pull request by robfletcher**
To help us pin down problematic apps.

### CODE REVIEW

File: keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionCheckerTests.kt [hunk 1]
```diff
@@ -174,7 +174,7 @@ internal class NewEnvironmentPromotionCheckerTests : JUnit5Minutests {
 
           test("an event is not sent") {
             verify(exactly = 0) {
-              publisher.publishEvent(any<ArtifactVersionApproved>())
+              publisher.publishEvent(ofType<ArtifactVersionApproved>())
```
Comment (robfletcher) [inline keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionCheckerTests.kt:177]: Turns out that's not how you do it

Comment (emjburns) [reply keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionCheckerTests.kt:177]: oh interesting

Comment (robfletcher) [reply keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionCheckerTests.kt:177]: Yeah, I fixed a bunch more in https://github.com/spinnaker/keel/pull/1631

Comment (robfletcher) [reply keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionCheckerTests.kt:177]: `any()` is fine in some situations, but if you want to strictly match the type of the parameter that actually gets passed, that's not what it does.

**Review by lorin — APPROVED** [review]
Nice and elegant.

**Pull request merged**